java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -115,11 +120,61 @@ private void run(@Nullable final ExplodedGraph.Node currentNode, @Nullable final
       .collect(Collectors.toList());
 
     if (learnedConstraints.stream().anyMatch(addToFlow)) {
-      flow.add(location(currentNode.parent()));
+      ExplodedGraph.Node parentNode = currentNode.parent();
+      if (parentNode.programPoint.syntaxTree().is(Tree.Kind.METHOD_INVOCATION) && currentNode.programState.selectedYield() != null) {
+        flowFromMethodInvocation(currentNode.programState.selectedYield(), parentNode);
+      } else {
+        flow.add(location(parentNode));
+      }
     }
     return learnedConstraints;
   }
 
+  private void flowFromMethodInvocation(MethodYield selectedYield, ExplodedGraph.Node parentNode) {
+    List<JavaFileScannerContext.Location> flowFromMethodInvocation;
+    String message;
+    Symbol argumentSymbol = correspondingArgumentSymbol(symbolicValue, parentNode);
+    if (selectedYield.exception) {
```

Comment:
All this could probably be simplified to have only one call to Flowcomputation with the proper arguments